clinica/iotools/bids_utils.py

```diff
@@ -467,15 +467,20 @@ def _write_bids_dataset_description(
         bids_desc.write(to=f)
 
 
-def _write_readme(bids_dir: Union[str, Path]) -> None:
-    """Write `README`file at the root of the BIDS directory."""
-    import clinica
+def _write_readme(
+    study_name: str,
+    data_dict: dict,
+    bids_dir: Union[str, Path],
+    bids_version: Optional[str] = None,
+) -> None:
+    """Write `dataset_description.json` at the root of the BIDS directory."""
+    from clinica.iotools.bids_readme import BIDSReadme
 
+    bids_desc = BIDSReadme(
+        name=study_name, link=data_dict["link"], desciption=data_dict["desc"]
+    )
     with open(Path(bids_dir) / "README", "w") as f:
-        f.write(
-            f"This BIDS directory was generated with Clinica v{clinica.__version__}.\n"
-            f"More information on https://www.clinica.run\n"
-        )
+        bids_desc.write(to=f)
```

Comment:
This feels a little strange. I'd expect a `write` method which would take a file name as input and the method would handle the file opening and closing. WDYT?

Reply:
Traditionnally, `read` / `write` are for buffers, `load` / `save` for files. Agreed, the scientific community mixed both, hence the potential for confusion.

Having `write` handle opening and closing the file improves usage convenience in exchange for additionnal checks (is the path valid and writeable), resource management and reduced testability. If `write` only accepts buffers, then responsibilities are lifted to the caller, but the implementation can be easility tested in memory.

The other alternative, which most libraries do, is to accept both (a buffer or a path). Since this code is only used internally by Clinica, I'd favour robustness and testability other convenience, but opinions may vary 👍

Reply:
I'm not really convinced by the argument that this is only used internally. AFAICT the `BIDSReadme` class is public as well as its `write` method. Even if most users won't have to use this class, some future contributors might.
I'm fine with both approaches, but as you wrote, most scientific libraries (Pandas, Numpy...) traditionally accept buffer and path with loading/reading/saving/writing methods.
If we decide to implement an uncommon design by only supporting buffers, I think we should at least document it better as most users and contributors will probably expect the usual buffer+path behavior.

Reply:
> I'm not really convinced by the argument that this is only used internally.

I meant Clinica is not used as a library, so none of these interfaces are for public consumption. Unlike say the `numpy` and `pandas` loaders and writers.

> If we decide to implement an uncommon design by only supporting buffers.

I would not label this as uncommon. `json.load` from the standard library works on file objects only.

Reply:
Alright, I'll keep it that way for now.